apps/balancer-tools/src/app/poolsimulator/(components)/ImpactCurve.tsx

```diff
@@ -156,6 +171,24 @@ export function ImpactCurve() {
     ),
   ];
 
+  const maxOfInitial = Math.max(
+    ...initialAmountsAnalysisTokenIn,
+    ...initialAmountsTabTokenIn,
+  );
+  const maxOfCustom = Math.max(
+    ...variantAmountsAnalysisTokenIn,
+    ...variantAmountsTabTokenIn,
+  );
+  const xlimit = Math.min(maxOfInitial, maxOfCustom);
+  const indexOfXLimit = initialAmountsAnalysisTokenIn.indexOf(xlimit);
```

Comment:
```suggestion
  const xLimitIndex = initialAmountsAnalysisTokenIn.indexOf(xlimit);
```
also, would this always yield a result or could it be `-1`? 🤔 xLimit could be a value that's not in `initialAmountsAnalysisTokenIn` right?

Reply:
you are right! fixing rn

Reply:
done here https://github.com/bleu-studio/balancer-tools/pull/183/commits/1beba2c801689c8e75d149f14dfd3ad4834829dc